File: src/main/java/io/curiousoft/ijudi/ordermanagement/validator/ValidMobileNumber.java
package io.curiousoft.ijudi.ordermanagement.validator;

import javax.validation.Payload;
import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.*;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Target({ FIELD, TYPE, PARAMETER })
@Retention(RUNTIME)
@Documented
public @interface ValidMobileNumber {

    String message() default "Mobile not format is not valid. Please put like +27812815577 or 27812815577";

    Class<?>[] groups() default { };

    Class<? extends Payload>[] payload() default { };
}

File: izinga-commons/src/main/kotlin/io/curiousoft/izinga/commons/validator/ValidMobileNumber.kt
package io.curiousoft.izinga.commons.validator

import java.lang.annotation.Documented
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy
import javax.validation.Payload
import kotlin.reflect.KClass

@Target(
    AnnotationTarget.FIELD,
    AnnotationTarget.ANNOTATION_CLASS,
    AnnotationTarget.CLASS,
    AnnotationTarget.VALUE_PARAMETER
)
@Retention(
    RetentionPolicy.RUNTIME
)
@Documented
annotation class ValidMobileNumber(
    val message: String = "Mobile not format is not valid. Please put like +27812815577 or 27812815577",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = []
)